tensor_comprehensions/pybinds/tclib.cc

```diff
@@ -0,0 +1,498 @@
+            ScopeGuard sg([&config]() { config.__exit__(); });
+            std::vector<at::Tensor> atInputs = getATenTensors(inputs);
+            auto bestOptions = instance.tune(
+                entryPoint, atInputs, baseMapping, instance.fileName);
+            if (bestOptions.size() > 0u) {
+              return bestOptions[0];
+            } else {
+              std::cout << "Autotuner could not find options, returning base"
+                        << std::endl;
+              return baseMapping;
+            }
+          });
+
+  py::class_<CudaMappingOptionsCache>(
+      m, "MappingOptionsCache", py::module_local())
+      .def(py::init<std::string>())
+      .def(
+          "load",
+          [](CudaMappingOptionsCache& instance,
+             const std::string& tc,
+             const std::string& entryPoint,
+             py::tuple& inputs,
+             const size_t& numCandidates) {
+            return instance.load(
+                tc, entryPoint, getATenTensors(inputs), numCandidates);
+          });
+
+  py::class_<CudaCompilationUnit>(m, "CompilationUnit", py::module_local())
+      .def(py::init<std::string>())
+      .def(
+          "is_compiled",
+          [](CudaCompilationUnit& instance,
+             const std::string& entryPoint,
+             py::tuple& inputs) { return false; })
+      .def(
+          "inferOutputs",
+          [](CudaCompilationUnit& instance,
+             const std::string& entryPoint,
+             py::tuple& inputs) {
+            auto atInputs = getATenTensors(inputs);
+            auto atOutputs =
+                tc::aten::prepareOutputs(instance.tc, entryPoint, atInputs);
+            return convertToPyObjects(atOutputs);
+          })
+      .def(
+          "compile",
+          [](CudaCompilationUnit& instance,
+             const std::string& entryPoint,
+             py::tuple& inputs,
+             const tc::CudaMappingOptions& options) {
+            instance.compiled[entryPoint] = tc::aten::compile<tc::CudaBackend>(
+                instance.tc, entryPoint, getATenTensors(inputs), options);
+          })
+      .def(
+          "run",
+          [](CudaCompilationUnit& instance,
+             const std::string& entryPoint,
+             py::tuple& inputs,
+             py::tuple& outputs) {
+            CHECK_GE(outputs.size(), 1u)
+                << "run needs a tuple of output tensors to write into";
+            auto atInputs = getATenTensors(inputs);
+            auto atOutputs = getATenTensors(outputs);
+            tc::aten::run(
+                *instance.compiled.at(entryPoint), atInputs, atOutputs);
+          })
+      .def(
+          "uncheckedRun",
+          [](CudaCompilationUnit& instance,
+             const std::string& entryPoint,
+             py::tuple& inputs,
+             py::tuple& outputs) {
+            CHECK_GE(outputs.size(), 1u)
+                << "uncheckedRun needs a tuple of output tensors to write into";
+            tc::aten::uncheckedRun(
```

Comment:
Why can't you just call `instace.compiled.at(entryPoint)->uncheckedRun(...)` here?  There's no need for that free function in `tc::aten`.